cranelift/filetests/src/function_runner.rs

```diff
@@ -66,27 +66,47 @@ impl SingleFunctionCompiler {
     ///  - compile the [Function]
     ///  - compile a `Trampoline` for the [Function]'s signature (or used a cached `Trampoline`;
     ///    this makes it possible to call functions when the signature is not known until runtime.
-    pub fn compile(&mut self, function: Function) -> Result<CompiledFunction, CompilationError> {
+    pub fn compile(self, function: Function) -> Result<CompiledFunction, CompilationError> {
         let signature = function.signature.clone();
         if signature.call_conv != self.isa.default_call_conv() {
             return Err(CompilationError::InvalidTargetIsa);
         }
 
-        // Compile the function itself.
-        let code_page = compile(function, self.isa.as_ref())?;
-
-        // Compile the trampoline to call it, if necessary (it may be cached).
-        let isa = self.isa.as_ref();
-        let trampoline = self
-            .trampolines
-            .entry(signature.clone())
-            .or_insert_with(|| {
-                let ir = make_trampoline(&signature, isa);
-                let code = compile(ir, isa).expect("failed to compile trampoline");
-                Trampoline::new(code)
-            });
-
-        Ok(CompiledFunction::new(code_page, signature, trampoline))
+        let trampoline = make_trampoline(&signature, self.isa.as_ref());
+
+        let builder = JITBuilder::with_isa(self.isa, cranelift_module::default_libcall_names());
+        let mut module = JITModule::new(builder);
+        let mut ctx = module.make_context();
+
+        let name = format!("{}", function.name);
+        let func_id = module.declare_function(&name, Linkage::Local, &function.signature)?;
+
+        // Build and declare the trampoline in the module
+        let trampoline_name = format!("{}", trampoline.name);
+        let trampoline_id =
+            module.declare_function(&trampoline_name, Linkage::Local, &trampoline.signature)?;
```

Comment:
If it works out to keep the `SingleFunctionCompiler` reusable, do you think it'd be worth keeping the cache of trampolines around too?